File: TerzulliFileManager/app/src/main/java/com/terzulli/terzullifilemanager/database/daos/ItemDao.java
package com.terzulli.terzullifilemanager.database.daos;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Update;

import com.terzulli.terzullifilemanager.database.entities.TableItem;

@Dao
public interface ItemDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(TableItem item);

    @Update
    void update(TableItem item);

    @Delete
    void delete(TableItem item);
}

File: TerzulliFileManager/app/src/main/java/com/terzulli/terzullifilemanager/database/daos/ItemDao.kt
package com.terzulli.terzullifilemanager.database.daos

import androidx.room.*
import com.terzulli.terzullifilemanager.database.entities.TableItem

@Dao
interface ItemDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(item: TableItem?): Long

    @Update
    fun update(item: TableItem?)

    @Delete
    fun delete(item: TableItem?)
}